src/lib.rs

```diff
@@ -61,14 +60,17 @@ impl Default for Optimizer {
 }
 
 impl Optimizer {
-    pub fn parse_and_optimize(&mut self, program: &str) -> Result<Program, EggCCError> {
+    pub fn parse_and_optimize(
+        &mut self,
+        program: &str,
+    ) -> Result<Program, Box<dyn std::error::Error>> {
```

Comment:
I don't think this should be a `std::error::Error`. The benefit of the `EggCCError` is we control the output- we want to output good error messages that have meaning to the user of eggcc, not random rust errors